File: example-question/python/solutions/aggregate.py
#!/bin/python

import math
import os
import random
import re
import sys

def AggregateNames(nameList):
    aggregated = {}
    
    for name in nameList :
        if name in aggregated.keys():
            aggregated[name] += 1
        else:
            aggregated[name] = 1
    return aggregated


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    nameList_count = int(raw_input().strip())

    nameList = []

    for _ in xrange(nameList_count):
        nameList_item = raw_input()
        nameList.append(nameList_item)

    result = AggregateNames(nameList)
    for key in sorted(result.keys()):
        fptr.write(str(key) + ' ' + str(result[key]) + '\n')

    fptr.close()

File: example-question/java/solutions/Aggregate.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

class Result {
    public static Map<String, Integer> AggregateNames(String[] nameList) {
        Map<String, Integer> map = new TreeMap<String, Integer>();
        Integer ONE = Integer.valueOf(1);

        for (int i = 0, n = nameList.length; i < n; i++) {
            String name = nameList[i];
            if (map.containsKey(name))
                map.put(name, map.get(name) + 1);
            else
                map.put(name,ONE);
        }
        return map;
    }
}


public class Aggregate {
    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(System.in);
        final String fileName = System.getenv("OUTPUT_PATH");
        BufferedWriter bw = new BufferedWriter(new FileWriter(fileName));
        Map<String, Integer> res = new TreeMap<String, Integer>();
        
        int _nameList_size = 0;
        _nameList_size = Integer.parseInt(in.nextLine().trim());
        String[] _nameList = new String[_nameList_size];
        String _nameList_item;
        for(int _nameList_i = 0; _nameList_i < _nameList_size; _nameList_i++) {
            try {
                _nameList_item = in.nextLine();
            } catch (Exception e) {
                _nameList_item = null;
            }
            _nameList[_nameList_i] = _nameList_item;
        }
        
        res = Result.AggregateNames(_nameList);
        
        for (Map.Entry<String, Integer> entry : res.entrySet()) {
            bw.write(entry.getKey() + " " + String.valueOf(entry.getValue()));
            bw.newLine();
        }
        bw.close();
    }
}
